cloudformation/custom_resource/cluster.yaml

```diff
@@ -0,0 +1,225 @@
+      TimeoutInMinutes: 10
+      Parameters:
+        EnableIamAdminAccess: true
+
+  PclusterCfnFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/lambda/${PclusterCfnFunction}
+
+  EventsPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: EventsPolicy
+            Effect: Allow
+            Action:
+              - events:PutRule
+              - events:DeleteRule
+              - events:PutTargets
+              - events:RemoveTargets
+            Resource: !Sub arn:${AWS::Partition}:events:${AWS::Region}:${AWS::AccountId}:rule/*
+
+  PclusterLambdaRole:
+    Condition: UsePCPolicies
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRole
+            Principal:
+              Service: lambda.amazonaws.com
+      ManagedPolicyArns: !Split
+        - ","
+        - !Sub
+          - ${LambdaExecutionPolicy},${ClusterPolicy},${DefaultAdminPolicy},${EventsPolicy}${AdditionalIamPolicies}
+          - { LambdaExecutionPolicy: !Sub "arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+              ClusterPolicy: !GetAtt [ PclusterPolicies, Outputs.ParallelClusterClusterPolicy ],
+              DefaultAdminPolicy: !GetAtt [ PclusterPolicies, Outputs.DefaultParallelClusterIamAdminPolicy ],
+              EventsPolicy: !Ref EventsPolicy,
+              AdditionalIamPolicies: !If [UseAdditionalIamPolicies, !Sub [",${AdditionalPolicies}", AdditionalPolicies: !Join [',', !Ref AdditionalIamPolicies ]], ''] }
+
+  PclusterCfnFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub
+        - pcluster-cfn-${StackIdSuffix}
+        - { StackIdSuffix: !Select [2, !Split ['/', !Ref 'AWS::StackId']] }
+      TracingConfig:
+        Mode: Active
+      MemorySize: 1024
+      Timeout: 60
+      Code:
+        ZipFile: |
+          import datetime
+          import json
+          import logging
+          import random
+          import re
+          import string
+          import sys
+          logger = logging.getLogger()
+          logger.setLevel(logging.INFO)
+
+          import pcluster.api.controllers.cluster_operations_controller
+          import pcluster.api.errors
+          import pcluster.utils
+          from pcluster.api import encoder
+          from pcluster.cli.exceptions import APIOperationException, ParameterException
+          from pcluster.api.errors import exception_message, NotFoundException
+          import pcluster.lib as pc
+
+          crhelper_path = "/opt/python//pcluster/resources/custom_resources/custom_resources_code"
+          sys.path.insert(0, crhelper_path)
+          from crhelper import CfnResource
+          helper = CfnResource()
+
+          def update_response(response):
```

Comment:
why do we need to filter out some keys?

Reply:
I thought this was needed but double checked and it isn't, removed.